modin/core/dataframe/algebra/groupby.py

```diff
@@ -373,16 +412,196 @@ def try_filter_dict(agg_func, df):
             Aggregation function.
         df : pandas.DataFrame
             Serialized partition which contains available columns.
+        preserve_aggregation_order : bool, default: True
+            Whether to manually restore the order of columns for the result specified
+            by the `agg_func` keys (only makes sense when `agg_func` is a dictionary).
 
         Returns
         -------
-        Callable
+        callable(pandas.core.groupby.DataFrameGroupBy) -> pandas.DataFrame
             Aggregation function that can be safely applied to this particular partition.
         """
         if not isinstance(agg_func, dict):
             return agg_func
-        partition_dict = {k: v for k, v in agg_func.items() if k in df.columns}
-        return lambda grp: grp.agg(partition_dict)
+
+        grp_has_id_level = df.columns.names[0] == cls.ID_LEVEL_NAME
+        # The 'id' level prevents us from a lookup for the original
+        # partition's columns. So dropping the level.
+        partition_columns = df.columns.droplevel(0) if grp_has_id_level else df.columns
```

Comment:
```suggestion
        partition_columns = set(df.columns.droplevel(0) if grp_has_id_level else df.columns)
```
for huge frames the lookup (when building the dict below) should be faster in a set. Also can use `frozenset()` to save a teeny-tinsy bit of memory (but it could make `black` split the line so it would look ugly, and it doesn't give a lot, so pick whatever you like).